MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -0,0 +1,192 @@
+            
+            mapviewExpectation.fulfill()
+        }
+        
+        wait(for: [mapviewExpectation], timeout: delay)
+    }
+    
+    // Brief: navigationViewController(_:roadNameAt:) delegate method is implemented,
+    //        with a blank road name (empty string) provided and wayNameView label is hidden.
+    
+    func testNavigationViewControllerDelegateRoadNameAtLocationEmptyString() {
+        
+        let navigationViewController = dependencies.navigationViewController
+        let routeController = navigationViewController.routeController!
+        
+        // Identify a location to set the custom road name.
+        let turkStreetLocation = dependencies.poi[1]
+        let roadName = ""
+        customRoadName[turkStreetLocation.coordinate] = roadName
+        
+        let mapviewExpectation = expectation(description: "mapview_exp_hide_roadname")
+        let customMapView = MGLMapView(frame: (navigationViewController.mapView?.bounds)!, styleURL: URL(string: "mapbox://styles/mapbox/streets-v9"))
+        
+        let delay = 10.0
+        let fireTime = DispatchTime.now() + delay
+        
+        DispatchQueue.main.asyncAfter(deadline: fireTime) {
+            
+            navigationViewController.mapView?.delegate?.mapView?(customMapView, didFinishLoading: customMapView.style!)
+            navigationViewController.mapView?.delegate?.mapViewDidFinishLoadingMap?(customMapView)
+            
+            routeController.locationManager(routeController.locationManager, didUpdateLocations: [turkStreetLocation])
+            
+            let wayNameView = (navigationViewController.mapViewController?.navigationView.wayNameView)!
+            let currentRoadName = wayNameView.text!
+            XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+            XCTAssertTrue(wayNameView.isHidden, "WayNameView should be hidden.")
+            
+            mapviewExpectation.fulfill()
+        }
+        
+        wait(for: [mapviewExpectation], timeout: delay)
+    }
+    
+    // TODO: Most likely a UI-Test candidate.
+    func testNavigationViewControllerDelegateRoadNameAtLocationUmimplemented() {
+        
+        let dummyNavigationViewController = dependencies.dummyNavigationViewController
+        let routeController = dummyNavigationViewController.routeController!
+        
+        // Identify a location without a custom road name.
+        let fultonStreetLocation = dependencies.poi[2]
+        
+        let mapviewExpectation = expectation(description: "mapview_exp_default_roadname")
+        let customMapView = MGLMapView(frame: (dummyNavigationViewController.mapView?.bounds)!, styleURL: URL(string: "mapbox://styles/mapbox/streets-v9"))
+        
+        let delay = 1.0
+        let fireTime = DispatchTime.now() + delay
+        
+        DispatchQueue.main.asyncAfter(deadline: fireTime) {
```

Comment:
This is an unconventional way to set up an expectation regarding a “did finish” delegate method. For one thing, it subjects the test to a hard-coded time interval to delay the test by. Is this test expected to actually load vector tiles to query? If so, should any of that process be mocked? If not, is a delay necessary?

Normally, an integration test would cause the delegate method to fire organically – in this case, by setting a style – and set an expectation that gets fulfilled inside an implementation of the delegate method. The point is to allow the `mapViewDidFinishLoadingMap(_:)` method to be called on its own; calling that method directly would only test the implementation of that method, which isn’t the focus of this test.